src/Common/tests/System/Collections/IDictionary.Generic.Tests.netcoreapp1.1.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace System.Collections.Tests
+{
+    /// <summary>
+    /// Contains tests that ensure the correctness of any class that implements the generic
+    /// IDictionary interface
+    /// </summary>
+    public abstract partial class IDictionary_Generic_Tests<TKey, TValue> : ICollection_Generic_Tests<KeyValuePair<TKey, TValue>>
+    {
+        #region KeyValuePair
+
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void KeyValuePair_Deconstruct(int count)
+        {
+            IDictionary<TKey, TValue> dictionary = GenericIDictionaryFactory(count);
+
+            Assert.All(dictionary, KeyValuePair_Deconstruct);
```

Comment:
It can be used to test deconstruction in other methods. For example I can create test for empty pair deconstruction and use this method in new test. I did not create such testing because it seems useless, but it is good to split one methods to logical parts.